tools/c7n_gcp/c7n_gcp/resources/loadbalancer.py

```diff
@@ -56,6 +57,70 @@ def get(client, resource_info):
                 'urlMap': resource_info['name']})
 
 
+@LoadBalancingUrlMap.action_registry.register('delete')
+class LoadBalancingUrlMapDelete(MethodAction):
+    """The action is used for Load Balancing URL Maps delete.
+    GCP action is https://cloud.google.com/compute/docs/reference/rest/v1/urlMaps/delete.
+
+    Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-loadbalancer-url-map-delete
+            resource: gcp.loadbalancer-url-map
+            filters:
+              - type: value
+                key: name
+                op: contains
+                value: url-map
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    method_spec = {'op': 'delete'}
+
+    def get_resource_params(self, model, resource):
+        project = local_session(self.manager.source.query.session_factory).get_default_project()
+        return {
+            'project': project,
+            'urlMap': resource['name']}
+
+
+@LoadBalancingUrlMap.action_registry.register('invalidate-cache')
+class LoadBalancingUrlMapInvalidateCache(MethodAction):
+    """The action is used for Load Balancing URL Maps cache invalidating.
```

Comment:
what is the use case for this, this feels like application deployment functionality that belongs with the application.

Reply:
The use case can be following. There are some load balancers for some applications that uses one of databases / resources. There was external change in the database / other resources, but load balancers saved state based on previous state. The use case helps to avoid using inconsistent data in this case

Reply:
that relies on app specific knowledge, an infrastructure team across projects can't know that safely based on strings alone. ie the functionality here alone doesn't enable detection ofnstale entries, which means its squarely in the domain of the application team.

Reply:
@kapilt should we remove invalidate-cache action?

Reply:
yeah i thinks its a good idea to leave it off and just have this pr with delete

Reply:
invalidate-cache is deleted

Reply:
@kapilt can we improve something in the PR?

Reply:
Hiya @kapilt, anything else here?

Reply:
@kapilt Tests seem to be fixed. Can this be merged?